recipe/test-builtin-extensions.sh

```diff
@@ -3,9 +3,10 @@ set -euxo pipefail
 PKG_PREFIX='duckdb-extension-'
 EXTENSION_NAME="${PKG_NAME#$PKG_PREFIX}"
 
-QUERY="select extension_name from duckdb_extensions() where (installed and install_mode = 'STATICALLY_LINKED');"
+# jemalloc is statically linked into DuckDB only on the Linux platform. Skip it here for simplicity.
+QUERY="select extension_name from duckdb_extensions() where (installed and install_mode = 'STATICALLY_LINKED' and extension_name <> 'jemalloc');"
```

Comment:
We should make the test depend on the OS instead.

Reply:
I am not sure how to do it right